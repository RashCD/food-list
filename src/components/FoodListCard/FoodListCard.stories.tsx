import type { Meta, StoryObj } from '@storybook/react';

import FoodListCard from './FoodListCard';

const meta = {
	title: 'Components/FoodListCard',
	component: FoodListCard,
	parameters: {
		layout: 'centered',
	},
	tags: ['autodocs'],
	argTypes: {},
} satisfies Meta<typeof FoodListCard>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
	args: {
		foodList: [
			{
				id: '628b5decc94a27754f30e6f1',
				index: 0,
				rating: 3.9508,
				promotion: 'gift',
				isNew: false,
				categoryId: '6288a89fac9e970731bfaa7b',
				minCookTime: 80,
				maxCookTime: 100,
				restaurant: 'Niquent',
				name: 'Niquent Drinks',
				imageUrl: 'https://source.unsplash.com/random/400x400?Drinks',
			},
			{
				id: '628b5decf39bcc4e982fc88a',
				index: 1,
				rating: 4.9874,
				promotion: '1+1',
				isNew: false,
				categoryId: '6288a89f1f0152b8c2cd512b',
				minCookTime: 120,
				maxCookTime: 140,
				restaurant: 'Boilicon',
				name: 'Boilicon Shushi',
				imageUrl: 'https://source.unsplash.com/random/400x400?Shushi',
			},
			{
				id: '628b5dec6678e96d75f2f7de',
				index: 2,
				rating: 3.4518,
				promotion: null,
				isNew: true,
				categoryId: '6288a89f1f0152b8c2cd512b',
				minCookTime: 100,
				maxCookTime: 120,
				restaurant: 'Quinex',
				name: 'Quinex Shushi',
				imageUrl: 'https://source.unsplash.com/random/400x400?Shushi',
			},
			{
				id: '628b5dec97eacf5e8a604bd7',
				index: 3,
				rating: 1.5975,
				promotion: 'gift',
				isNew: false,
				categoryId: '6288a89f7338764f2071a8a8',
				minCookTime: 120,
				maxCookTime: 140,
				restaurant: 'Perkle',
				name: 'Perkle Pizza',
				imageUrl: 'https://source.unsplash.com/random/400x400?Pizza',
			},
			{
				id: '628b5decf99b6a8dc80af3b6',
				index: 4,
				rating: 0.8644,
				promotion: null,
				isNew: true,
				categoryId: '6288a89fac9e970731bfaa7b',
				minCookTime: 70,
				maxCookTime: 90,
				restaurant: 'Zanymax',
				name: 'Zanymax Drinks',
				imageUrl: 'https://source.unsplash.com/random/400x400?Drinks',
			},
			{
				id: '628b5dec0690be0f73109de7',
				index: 5,
				rating: 4.7915,
				promotion: '1+1',
				isNew: true,
				categoryId: '6288a89fe6c2fe0b758360fe',
				minCookTime: 90,
				maxCookTime: 110,
				restaurant: 'Sunclipse',
				name: 'Sunclipse Deserts',
				imageUrl: 'https://source.unsplash.com/random/400x400?Deserts',
			},
		],
	},
};
