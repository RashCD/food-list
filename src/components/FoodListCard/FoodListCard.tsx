import { FoodListAPITypes } from '../../pages/Home/constant';
import FoodCard from '../FoodCard/FoodCard';
import styles from './foodlistcard.module.css';

type FoodListCardProps = {
	foodList: FoodListAPITypes['foods'];
};

const FoodListCard = ({ foodList }: FoodListCardProps) => {
	if (foodList.length === 0) {
		return (
			<div className={styles.noResult}>
				<p>No Result</p>
			</div>
		);
	}

	return (
		<div className={styles.foodList}>
			{foodList.map((food) => (
				<FoodCard
					key={food.id}
					name={food.name}
					rating={food.rating}
					minCookTime={food.minCookTime}
					maxCookTime={food.maxCookTime}
					isNew={food.isNew}
					promotion={food.promotion}
					imageUrl={food.imageUrl}
				/>
			))}
		</div>
	);
};

export default FoodListCard;
