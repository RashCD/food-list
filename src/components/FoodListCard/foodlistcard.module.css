.noResult {
	color: slategray;
	font-weight: 500;
	text-align: center;
}

.foodList {
	display: grid;
	grid-template-columns: repeat(4, minmax(300px, 1fr));
	gap: 1rem;
	align-self: center;

	@media only screen and (max-width: 1200px) {
		grid-template-columns: repeat(3, minmax(300px, 1fr));
	}

	@media only screen and (max-width: 930px) {
		grid-template-columns: repeat(2, minmax(0, 1fr));
	}

	@media only screen and (max-width: 600px) {
		grid-template-columns: repeat(1, minmax(0, 1fr));
	}
}
