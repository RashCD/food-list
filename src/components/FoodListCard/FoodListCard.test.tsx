import { render, screen } from '@testing-library/react';
import FoodListCard from './FoodListCard';

describe('FoodListCard component', () => {
	it('should render', () => {
		render(
			<FoodListCard
				foodList={[
					{
						id: '628b5decc94a27754f30e6f1',
						index: 0,
						rating: 3.9508,
						promotion: 'gift',
						isNew: false,
						categoryId: '6288a89fac9e970731bfaa7b',
						minCookTime: 80,
						maxCookTime: 100,
						restaurant: 'Niquent',
						name: 'Niquent Drinks',
						imageUrl:
							'https://source.unsplash.com/random/400x400?Drinks',
					},
					{
						id: '628b5decf39bcc4e982fc88a',
						index: 1,
						rating: 4.9874,
						promotion: '1+1',
						isNew: false,
						categoryId: '6288a89f1f0152b8c2cd512b',
						minCookTime: 120,
						maxCookTime: 140,
						restaurant: 'Boilicon',
						name: 'Boilicon Shushi',
						imageUrl:
							'https://source.unsplash.com/random/400x400?Shushi',
					},
				]}
			/>,
		);

		expect(screen.getByText('Niquent Drinks')).toBeInTheDocument();
		expect(screen.getByText('Boilicon Shushi')).toBeInTheDocument();
	});

	it('should display 2 main icon when foodList is provided', () => {
		render(
			<FoodListCard
				foodList={[
					{
						id: '628b5decc94a27754f30e6f1',
						index: 0,
						rating: 3.9508,
						promotion: 'gift',
						isNew: false,
						categoryId: '6288a89fac9e970731bfaa7b',
						minCookTime: 80,
						maxCookTime: 100,
						restaurant: 'Niquent',
						name: 'Niquent Drinks',
						imageUrl:
							'https://source.unsplash.com/random/400x400?Drinks',
					},
					{
						id: '628b5decf39bcc4e982fc88a',
						index: 1,
						rating: 4.9874,
						promotion: 'discount',
						isNew: false,
						categoryId: '6288a89f1f0152b8c2cd512b',
						minCookTime: 120,
						maxCookTime: 140,
						restaurant: 'Boilicon',
						name: 'Boilicon Shushi',
						imageUrl:
							'https://source.unsplash.com/random/400x400?Shushi',
					},
				]}
			/>,
		);

		expect(screen.getByAltText('Gift icon')).toBeInTheDocument();
		expect(screen.getByAltText('Discount icon')).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const comp = render(
			<FoodListCard
				foodList={[
					{
						id: '628b5decc94a27754f30e6f1',
						index: 0,
						rating: 3.9508,
						promotion: 'gift',
						isNew: false,
						categoryId: '6288a89fac9e970731bfaa7b',
						minCookTime: 80,
						maxCookTime: 100,
						restaurant: 'Niquent',
						name: 'Niquent Drinks',
						imageUrl:
							'https://source.unsplash.com/random/400x400?Drinks',
					},
					{
						id: '628b5decf39bcc4e982fc88a',
						index: 1,
						rating: 4.9874,
						promotion: 'discount',
						isNew: false,
						categoryId: '6288a89f1f0152b8c2cd512b',
						minCookTime: 120,
						maxCookTime: 140,
						restaurant: 'Boilicon',
						name: 'Boilicon Shushi',
						imageUrl:
							'https://source.unsplash.com/random/400x400?Shushi',
					},
				]}
			/>,
		);

		expect(comp).toMatchSnapshot();
	});
});
