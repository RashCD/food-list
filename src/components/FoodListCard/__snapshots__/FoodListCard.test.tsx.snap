// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FoodListCard component > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_foodList_a8015e"
      >
        <div
          class="_card_2f15a9 _foodCardContainer_c0fa88"
        >
          <div
            class="_foodPromotionContainer_c0fa88 _gift_c0fa88"
          >
            <img
              alt="Gift icon"
              class="_icon_fa96f1 _promotionIcon_c0fa88"
              src="/src/assets/icons/gift.svg"
            />
          </div>
          <img
            alt="Niquent Drinks image"
            height="150"
            src="https://source.unsplash.com/random/400x400?Drinks"
            style="aspect-ratio: 2 / 1; object-fit: cover;"
          />
          <div
            class="_cardContent_2f15a9"
          >
            <div>
              <span>
                Niquent Drinks
              </span>
            </div>
            <div>
              <div
                class="_badge_58bfbf _secondary_58bfbf"
              >
                <img
                  alt="Niquent Drinks rating"
                  class="_icon_fa96f1 _icon_c0fa88"
                  src="/src/assets/icons/star.svg"
                />
                4.0
              </div>
              <div
                class="_badge_58bfbf _secondary_58bfbf"
              >
                80 - 100 min
              </div>
            </div>
          </div>
        </div>
        <div
          class="_card_2f15a9 _foodCardContainer_c0fa88"
        >
          <div
            class="_foodPromotionContainer_c0fa88 _discount_c0fa88"
          >
            <img
              alt="Discount icon"
              class="_icon_fa96f1 _promotionIcon_c0fa88"
              src="/src/assets/icons/discount.svg"
            />
          </div>
          <img
            alt="Boilicon Shushi image"
            height="150"
            src="https://source.unsplash.com/random/400x400?Shushi"
            style="aspect-ratio: 2 / 1; object-fit: cover;"
          />
          <div
            class="_cardContent_2f15a9"
          >
            <div>
              <span>
                Boilicon Shushi
              </span>
            </div>
            <div>
              <div
                class="_badge_58bfbf _secondary_58bfbf"
              >
                <img
                  alt="Boilicon Shushi rating"
                  class="_icon_fa96f1 _icon_c0fa88"
                  src="/src/assets/icons/star.svg"
                />
                5.0
              </div>
              <div
                class="_badge_58bfbf _secondary_58bfbf"
              >
                120 - 140 min
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_foodList_a8015e"
    >
      <div
        class="_card_2f15a9 _foodCardContainer_c0fa88"
      >
        <div
          class="_foodPromotionContainer_c0fa88 _gift_c0fa88"
        >
          <img
            alt="Gift icon"
            class="_icon_fa96f1 _promotionIcon_c0fa88"
            src="/src/assets/icons/gift.svg"
          />
        </div>
        <img
          alt="Niquent Drinks image"
          height="150"
          src="https://source.unsplash.com/random/400x400?Drinks"
          style="aspect-ratio: 2 / 1; object-fit: cover;"
        />
        <div
          class="_cardContent_2f15a9"
        >
          <div>
            <span>
              Niquent Drinks
            </span>
          </div>
          <div>
            <div
              class="_badge_58bfbf _secondary_58bfbf"
            >
              <img
                alt="Niquent Drinks rating"
                class="_icon_fa96f1 _icon_c0fa88"
                src="/src/assets/icons/star.svg"
              />
              4.0
            </div>
            <div
              class="_badge_58bfbf _secondary_58bfbf"
            >
              80 - 100 min
            </div>
          </div>
        </div>
      </div>
      <div
        class="_card_2f15a9 _foodCardContainer_c0fa88"
      >
        <div
          class="_foodPromotionContainer_c0fa88 _discount_c0fa88"
        >
          <img
            alt="Discount icon"
            class="_icon_fa96f1 _promotionIcon_c0fa88"
            src="/src/assets/icons/discount.svg"
          />
        </div>
        <img
          alt="Boilicon Shushi image"
          height="150"
          src="https://source.unsplash.com/random/400x400?Shushi"
          style="aspect-ratio: 2 / 1; object-fit: cover;"
        />
        <div
          class="_cardContent_2f15a9"
        >
          <div>
            <span>
              Boilicon Shushi
            </span>
          </div>
          <div>
            <div
              class="_badge_58bfbf _secondary_58bfbf"
            >
              <img
                alt="Boilicon Shushi rating"
                class="_icon_fa96f1 _icon_c0fa88"
                src="/src/assets/icons/star.svg"
              />
              5.0
            </div>
            <div
              class="_badge_58bfbf _secondary_58bfbf"
            >
              120 - 140 min
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
